src/main/java/com/bina/varsim/fastqLiftover/GenomeLocation.java

```diff
@@ -3,13 +3,20 @@
 public class GenomeLocation implements Comparable<GenomeLocation> {
     public String chromosome;
     public int location = 0;
+    public int read_location1 = -1; // if >=0, this is the 1-base index corresponding to a location in a read
```

Comment:
I've added class Position0 and Position1, both inherited from Position below. Picked Position because class Integer is already final'ed and can't served as base class. The clean way to do it in C++ is template<int base> class Integer; to do compile time check, can't figure out how to do it in java.